encoder.py

```diff
@@ -0,0 +1,144 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Optional, List, Dict, Any, Iterable
+
+import numpy as np
+
+from jina import Executor, requests, DocumentArray
+
+import torch
+import torch.nn as nn
+import torchvision.models as models
+
+
+def _batch_generator(data: List[Any], batch_size: int):
+    for i in range(0, len(data), batch_size):
+        yield data[i:min(i + batch_size, len(data))]
+
+
+class ImageTorchEncoder(Executor):
+    """
+    :class:`ImageTorchEncoder` encodes ``Document`` content from a ndarray,
+    potentially B x (Channel x Height x Width) into a ndarray of `B x D`.
```

Comment:
input should be Height, width, channel

Reply:
fixed, should be updated later with functions from jina-commons